easybuild/easyblocks/generic/systemmpi.py

```diff
@@ -0,0 +1,325 @@
+from easybuild.tools.toolchain import DUMMY_TOOLCHAIN_NAME
+
+class SystemMPI(Bundle, ConfigureMake, EB_impi):
+    """
+    Support for generating a module file for the system mpi with specified name.
+
+    The mpi compiler is expected to be available in $PATH, required libraries are assumed to be readily available.
+
+    Specifying 'system' as a version leads to using the derived mpi version in the generated module;
+    if an actual version is specified, it is checked against the derived version of the system mpi that was found.
+    """
+
+    @staticmethod
+    def extra_options():
+        """Add custom easyconfig parameters for SystemMPI easyblock."""
+        # Gather extra_vars from inherited classes, order matters to make sure bundle initialises correctly
+        extra_vars = ConfigureMake.extra_options()
+        extra_vars.update(EB_impi.extra_options())
+        extra_vars.update(Bundle.extra_options())
+        # Add an option to add all module path extensions to the resultant easyconfig
+        # This is useful if you are importing the MPI installation from a non-default path
+        extra_vars.update({
+            'generate_standalone_module': [False, "Add known path extensions and environment variables for the MPI "
+                                                  "installation to the final module", CUSTOM],
+        })
+        return extra_vars
+
+    def extract_ompi_setting(self, pattern, txt):
+        """Extract a particular OpenMPI setting from provided string."""
+
+        version_regex = re.compile(r'^\s+%s: (.*)$' % pattern, re.M)
+        res = version_regex.search(txt)
+        if res:
+            setting = res.group(1)
+            self.log.debug("Extracted OpenMPI setting %s: '%s' from search text", pattern, setting)
+        else:
+            raise EasyBuildError("Failed to extract OpenMPI setting '%s' using regex pattern '%s' from: %s",
+                                 pattern, version_regex.pattern, txt)
+
+        return setting
+
+    def __init__(self, *args, **kwargs):
+        """Extra initialization: determine system MPI version, prefix and any associated envvars."""
+        super(SystemMPI, self).__init__(*args, **kwargs)
+
+        mpi_name = self.cfg['name'].lower()
+
+        # Determine MPI wrapper path (real path, with resolved symlinks) to ensure it exists
+        if mpi_name == 'impi':
+            # For impi the version information is only found in *some* of the wrappers it ships, in particular it is
+            # not in mpicc
+            mpi_c_wrapper = 'mpiicc'
+            path_to_mpi_c_wrapper = which(mpi_c_wrapper)
+            if not path_to_mpi_c_wrapper:
+                mpi_c_wrapper = 'mpigcc'
+                path_to_mpi_c_wrapper = which(mpi_c_wrapper)
+                if not path_to_mpi_c_wrapper:
+                    raise EasyBuildError("Could not find suitable MPI wrapper to extract version for impi",
+                                         mpi_c_wrapper)
```

Comment:
this extra argument shouldn't be here, see https://github.com/ocaisa/easybuild-easyblocks/pull/11